app/src/main/java/fr/coppernic/scansample/MainActivity.java

```diff
@@ -265,4 +295,31 @@ private void displayServiceStatus(boolean start) {
             Timber.d("Barcode service stopped");
         }
     }
+
+    boolean scanIsStopped = true;
+    @Override
+    public boolean onKeyDown(int keyCode, KeyEvent keyEvent) {
+        // Change
+        Timber.d("onKeyDown()" + keyCode);
+        if (keyCode == OsHelper.getTriggerKey()) {
+            if (scanIsStopped) {
+                startScan();
+            }
+            scanIsStopped = false;
+            return false;
+        }
+        return super.onKeyDown(keyCode, keyEvent);
+    }
+    @Override
+    public boolean onKeyUp(int keyCode, KeyEvent keyEvent) {
+        Timber.d("onKeyUp()" + keyCode);
+        // Change
```

Comment:
why change? A comment saying that it is for Access trigger gun?

Reply:
Merci !